### ISSUE TITLE
Bug when loading biomaterial with .csv file

### ISSUE THREAD

**Issue by jhumann** [tripal/tripal_analysis_expression#275]
I am loading some biomaterials from a .csv file that I made with the following column headers: 
sample_name | description | biomaterial_provider | organism | cultivar | geo_loc_name | tissue | treatment

When I imported the .csv file with the Chado Biological Sample (Biomaterial) Loader, I got the following error:

2019-03-06 13:39:55: Calling: tripal_run_importer(10)

Running 'Chado Biological Sample (Biomaterial) Loader' importer
NOTE: Loading of file is performed using a database transaction.
If it fails or is terminated prematurely then all insertions and
updates are rolled back and will not be found in the database

Loading BioSample 1 of 6
Missing argument 6 for tripal_biomaterial_create_biomaterial(), called in                                                                          [warning]
/main/sites/hort-wd/dev.citrusgenomedb.org/drupal-7.54/sites/all/modules/tripal_analysis_expression/tripal_biomaterial/includes/TripalImporter/tripal_biomaterial_loader_v3.inc
on line 771 and defined tripal_biomaterial.api.inc:606

WD T_expression_load: PDOException: SQLSTATE[22P02]: Invalid text representation: 7 ERROR:  invalid input syntax for integer: "HLB susceptible     [error]
sample"
LINE 1: ...id, biosourceprovider_id) VALUES ('R19T23', '23', 'HLB susce...
                                                             ^: INSERT INTO chado.biomaterial (name, taxon_id, dbxref_id, biosourceprovider_id)
VALUES (:name, :taxon_id, :dbxref_id, :biosourceprovider_id); Array
(
    [:name] => R19T23
    [:taxon_id] => 23
    [:dbxref_id] => HLB susceptible sample
    [:biosourceprovider_id] => 714
)
 in chado_query() (line 1720 of
/main/sites/hort-wd/dev.citrusgenomedb.org/drupal-7.54/sites/all/modules/tripal/tripal_chado/api/tripal_chado.query.api.inc).
FAILED: Rolling back database changes...

Done.

Basically, it looks like it is trying to use "description" for the dbxref_id field.  The dbxref_id field is not a required column per the instructions on the loader page.  We did change the code and got the import to happen, but wanted to report this so it can get properly fixed.

Thanks,
Jodi

**Comment by bradfordcondon** [comment]
confirmed with our example file at https://github.com/tripal/tripal_analysis_expression/blob/master/example_files/exampleCSV.csv

Does your group need to use the flat file loader, or are the biomaterials hosted on NCBI?  I ask because I'm considering dropping support for the flat file loader.

**Comment by jhumann** [comment]
While I would love to just import NCBI biomaterial files, we have quite a few data sets that are not deposited in NCBI and loading using a .csv file is the most convenient for us.  It would be great if you could maintain this option.

**Comment by bradfordcondon** [comment]
I'm no longer sure I can reproduce this bug.  Would you please share your input file? 

Thank you.

**Comment by jhumann** [comment]
Here is the file we loaded.  Thanks!

[3175-biomaterial.csv.gz](https://github.com/tripal/tripal_analysis_expression/files/2957515/3175-biomaterial.csv.gz)

**Comment by bradfordcondon** [comment]
OK thanks.  This file also works for me locally and in the test case I wrote.

Is it possible theres an encoding issue, iie windows-specific characters?  Can you try using this copy of your file- https://github.com/tripal/tripal_analysis_expression/blob/275_flat_file/example_files/3175-biomaterial.csv

**Comment by chunhuaicheng** [comment]
I can confirm that the file loaded without a problem now. Thanks.
